table/block_based_table_reader.cc

```diff
@@ -1119,10 +1119,16 @@ bool BlockBasedTable::PrefixMayMatch(const Slice& internal_key) {
     return true;
   }
 
+  ParsedInternalKey parsed_internal_key;
+  if (!ParseInternalKey(internal_key, &parsed_internal_key)) {
+    // This shouldn't happen, but if it does just stumble forward.
+    return true;
+  }
+
   assert(rep_->ioptions.prefix_extractor != nullptr);
   auto prefix = rep_->ioptions.prefix_extractor->Transform(
-      ExtractUserKey(internal_key));
-  InternalKey internal_key_prefix(prefix, 0, kTypeValue);
+      parsed_internal_key.user_key);
+  InternalKey internal_key_prefix(prefix, parsed_internal_key.sequence, kTypeValue);
```

Comment:
I'm still wrapping my head around this, but do you think it would be more natural to have `kMaxSequenceNumber` here? That would also work for cases when index has (a, 3) and we're seeking for (a, 1). Even though the higher-level code will probably ignore (a, 3) if we're seeking for (a, 1), the name of this function is PrefixMayMatch() and it should return true in this case I think (since the prefix matches).